src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.NotebookTests/FunctionalTests.cs

```diff
@@ -0,0 +1,46 @@
+using System.Reflection;
+
+namespace GC.Infrastructure.NotebookTests
+{
+    public class FunctionalTests
+    {
+        [SetUp]
+        public void Setup()
+        {
+            if (!Utils.CheckDotnetReplInstalled())
+            {
+                Utils.InstallDotnetRepl();
+            }
+        }
+
+        [Test]
+        public void FunctionalTest_RunGCAnalysisExamples_Success()
+        {
+            // Conjecture: All examples should be functional.
+            string notebookPath = Path.Combine(Utils.GetNotebookDirectoryPath(), "Examples", "GCAnalysisExamples.ipynb");
+            Utils.RunNotebookThatsExpectedToPass(notebookPath);
+        }
+
+        [Test]
+        public void FunctionalTest_RunBenchmarkAnalysis_Success()
+        {
+            // TODO: Parameterize the notebook paths.
+            string notebookPath = Path.Combine(Utils.GetNotebookDirectoryPath(), "BenchmarkAnalysis.dib");
+            Utils.RunNotebookThatsExpectedToPass(notebookPath);
+        }
+
+        [Test]
+        public void FunctionalTest_IntentionallyFailedNotebook_Failure()
+        {
+            string? executionPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            Assert.IsNotNull(executionPath);
+            string failureNotebookPath = Path.Combine(executionPath, "TestNotebooks");
+            string[] failedNotebookPaths = Directory.GetFiles(failureNotebookPath);
+
+            foreach (var failedNotebookPath in failedNotebookPaths)
```

Comment:
This makes all of the "failed notebooks" a combined single test in test harness, so they can't be reported separately to the user.  Also if any of the individual notebooks throw an exception (not sure if a fluent assertion failure throws an exception or somehow calls into the nunit infra, but this would apply to any other uncaught exception), then the rest of the notebooks won't be tested.

Solutions could be to list the files (like `TestCase` above) or enumerate them in a `TestCaseSource`.  (Using more test features like this might be a reason to move over to xunit to match the runtime repo, but of course nunit is standard too.)